src/test/java/bio/terra/service/snapshot/SnapshotTest.java

```diff
@@ -285,4 +286,35 @@ public void snapshotByFullViewHappyPathTest() throws Exception {
         assertEquals("new snapshot has been created", snapshot.getName(), requestModel.getName());
         assertEquals("all 5 relationships come through", snapshot.getRelationships().size(), 5);
     }
+
+    @Test
+    public void snapshotInvalidEmailTest() throws Exception {
+        SnapshotRequestModel requestModel =
+            jsonLoader.loadObject("ingest-test-snapshot.json", SnapshotRequestModel.class);
+
+        String snapshotName = "mysnapshot12345";
+
+        requestModel.setName(snapshotName);
+        requestModel.setReaders(Collections.singletonList("bad-user@not-a-real-domain.com"));
+        DataRepoResponse<JobModel> jobResponse =
+            dataRepoFixtures.createSnapshotRaw(steward(), datasetSummaryModel.getName(), profileId, requestModel, false);
```

Comment:
Why does this test require a non-random name for the snapshot?

Reply:
It's because I want it to explicitly use the same name that it tried the first time (since that way it tests that the db was properly cleared)

Reply:
Wouldn't it be sufficient to test that the undostep no longer throws in the case where a snapshot no longer exists? This could be done with a mocked `IamService`.

Reply:
OK. I suppose you could let this API call generate a random name, and then request a non-random one below when you call `createSnapshotWithRequest()`.

Reply:
I don't think that would work since I want both to have the same name

Reply:
The first one's name will be random, the second will use the name generated by the first operation. Or is there no way to find out what name was generated by the first operation?

Reply:
Looking at `createSnapshotRaw`, it modifies the name of the model in place:
```
        requestModel.setName(Names.randomizeName(requestModel.getName()));
```
So the caller can look at `requestModel.getName()` to see what the random name is.